browser/ui/webui/brave_tip_ui.cc

```diff
@@ -331,5 +343,36 @@ void RewardsTipDOMHandler::OnRecurringTipSaved(
       "brave_rewards_tip.recurringTipSaved", base::Value(success));
 }
 
-BraveTipUI::~BraveTipUI() {
+void RewardsTipDOMHandler::TweetTip(const base::ListValue *args) {
+  DCHECK_EQ(args->GetSize(), 1U);
+  const base::DictionaryValue* tweet_metadata_dict = nullptr;
+  if (!args->GetDictionary(0, &tweet_metadata_dict))
+    return;
+
+  // Retrieve the relevant tweet metadata from arguments.
+  std::string screen_name;
+  if (!tweet_metadata_dict->GetString("screenName", &screen_name))
+    return;
+  std::string tweet_id;
+  if (!tweet_metadata_dict->GetString("tweetId", &tweet_id))
+    return;
+
+  // Construct the Twitter intent URL for the tip compliment.
+  std::string comment = l10n_util::GetStringFUTF8(
+      IDS_BRAVE_REWARDS_LOCAL_COMPLIMENT_TWEET, base::UTF8ToUTF16(screen_name));
+  std::string quoted_tweet_url =
+      base::StringPrintf("https://twitter.com/%s/status/%s",
+                         screen_name.c_str(), tweet_id.c_str());
```

Comment:
does this need to URL-encode the strings before inserting them into this URL?

Reply:
The GURL class handles it for you, but you just made me realize that I should be calling `gurl.is_valid()` before using the URL. Will add that now!